daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala

```diff
@@ -364,4 +365,40 @@ object Ref {
       assertRight(fromString(s))
   }
 
+  /** Package versions are non-empty strings consisting of segments of digits (without leading zeros)
+    *      separated by dots: "(0|[1-9][0-9]*)(\.(0|[1-9][0-9]*))*"
+    */
+  final case class PackageVersion private[data] (segments: ImmArray[Int])
+      extends Ordered[PackageVersion] {
+    override def toString: String = segments.toSeq.mkString(".")
+
+    override def compare(that: PackageVersion): Int = {
+      import scala.math.Ordering.Implicits._
+
+      implicitly[Ordering[Seq[Int]]].compare(segments.toSeq, that.segments.toSeq)
+    }
+  }
+
+  object PackageVersion {
+    private val MaxPackageVersionLength = 1024
+    final def fromString(s: String): Either[String, PackageVersion] = for {
+      _ <- Either.cond(
+        s.length <= MaxPackageVersionLength,
```

Comment:
We didn't have the size limit until now. Do you see a problem with introducing it now @remyhaemmerle-da ?

Reply:
I think limit  to 255 is fine.